src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/CertificatePal.PrivateKey.cs

```diff
@@ -80,6 +82,104 @@ public ECDsa GetECDsaPrivateKey()
             );
         }
 
+        public ICertificatePal CopyWithPrivateKey(DSA dsa)
+        {
+            DSACng dsaCng = dsa as DSACng;
+            ICertificatePal clone = null;
+
+            if (dsaCng != null)
+            {
+                clone = CopyWithPersistedCngKey(dsaCng.Key);
+
+                if (clone != null)
+                {
+                    return clone;
+                }
+            }
+
+            DSACryptoServiceProvider dsaCsp = dsa as DSACryptoServiceProvider;
+
+            if (dsaCsp != null)
+            {
+                clone = CopyWithPersistedCapiKey(dsaCsp.CspKeyContainerInfo);
+
+                if (clone != null)
+                {
+                    return clone;
+                }
+            }
+
+            DSAParameters privateParameters = dsa.ExportParameters(true);
```

Comment:
I assume you mean something opportunistic, like

```C#
DSAParameters privateParameters = dsa.ExportParameters(true);

fixed (byte* pinnedData = privateParameters.X)
{
    try
    {
        // do stuff
    }
    finally
    {
        Array.Clear(privateParameters.X);
    }
}
```

(alternatively, using GCHandle).  That won't be perfect, since the array could have been moved between the time it was filled and the time it was pinned; but it'll be like 5-nines correct.
```